### ISSUE TITLE
Crash after removing self-hosted site

### ISSUE THREAD

**Issue by rachelmcr** [wordpress-mobile/WordPress-iOS#11094]
I ran into this crash while testing the self-hosted logout flow (for #11093) in a simulator. I can reproduce it consistently when running the test `testSelfHostedUsernamePasswordLoginLogout` I implemented in that PR, although I can't reproduce it manually.

The test logs in to a self-hosted site in the app, and then removes the site. The crash happens after tapping the "Remove Site" option in the action sheet (the last step to remove the site from the app).

Stack trace:

```
* thread #1, queue = 'com.apple.main-thread', stop reason = breakpoint 1.1
    frame #0: 0x000000011740e705 libobjc.A.dylib`objc_exception_throw
    frame #1: 0x0000000117fe7015 CoreFoundation`+[NSException raise:format:] + 197
    frame #2: 0x0000000114bc6356 Foundation`-[NSRegularExpression(NSMatching) enumerateMatchesInString:options:range:usingBlock:] + 207
    frame #3: 0x0000000114bc6c8d Foundation`-[NSRegularExpression(NSMatching) matchesInString:options:range:] + 146
    frame #4: 0x0000000114bc71a2 Foundation`-[NSRegularExpression(NSReplacement) stringByReplacingMatchesInString:options:range:withTemplate:] + 285
  * frame #5: 0x000000010be1740d WordPress`-[Blog displayURL](self=0x000060000141e460, _cmd="displayURL") at Blog.m:143
    frame #6: 0x000000010be1786c WordPress`-[Blog hostURL](self=0x000060000141e460, _cmd="hostURL") at Blog.m:161
    frame #7: 0x000000010be1b51a WordPress`-[Blog supportsPluginManagement](self=0x000060000141e460, _cmd="supportsPluginManagement") at Blog.m:570
    frame #8: 0x000000010be1ac55 WordPress`-[Blog supports:](self=0x000060000141e460, _cmd="supports:", feature=BlogFeaturePluginManagement) at Blog.m:472
    frame #9: 0x000000010be6bd3b WordPress`-[BlogDetailsViewController configurationSectionViewModel](self=0x00007fd8885c0890, _cmd="configurationSectionViewModel") at BlogDetailsViewController.m:683
    frame #10: 0x000000010be68c99 WordPress`-[BlogDetailsViewController configureTableViewData](self=0x00007fd8885c0890, _cmd="configureTableViewData") at BlogDetailsViewController.m:534
    frame #11: 0x000000010c3084c2 WordPress`closure #1 in BlogDetailsViewController.startObservingQuickStart(notification=Foundation.Notification @ 0x00007ffee3e6cb68, self=0x00007fd8885c0890) at BlogDetailsViewController+FancyAlerts.swift:9
    frame #12: 0x000000010c3087bc WordPress`partial apply for closure #1 in BlogDetailsViewController.startObservingQuickStart() at <compiler-generated>:0
    frame #13: 0x000000010c007ae2 WordPress`thunk for @escaping @callee_guaranteed (@in_guaranteed Notification) -> () at <compiler-generated>:0
    frame #14: 0x0000000114aabd78 Foundation`-[__NSObserver _doit:] + 299
    frame #15: 0x0000000117f26bac CoreFoundation`__CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
    frame #16: 0x0000000117f2602f CoreFoundation`_CFXRegistrationPost + 447
    frame #17: 0x0000000117f25d71 CoreFoundation`___CFXNotificationPost_block_invoke + 225
    frame #18: 0x000000011800cae2 CoreFoundation`-[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1826
    frame #19: 0x0000000117f25694 CoreFoundation`_CFXNotificationPost + 964
    frame #20: 0x0000000114aac589 Foundation`-[NSNotificationCenter postNotificationName:object:userInfo:] + 66
    frame #21: 0x000000010c01f609 WordPress`QuickStartTourGuide.dismissCurrentNotice(self=0x0000600000258d80) at QuickStartTourGuide.swift:407
    frame #22: 0x000000010c020d3f WordPress`QuickStartTourGuide.endCurrentTour(self=0x0000600000258d80) at QuickStartTourGuide.swift:262
    frame #23: 0x000000010c5cc9f0 WordPress`QuickStartNavigationWatcher.navigationController(navigationController=0x00007fd8898eb400, viewController=0x00007fd8884beb90, animated=false, self=0x0000600003031020) at QuickStartNavigationWatcher.swift:14
    frame #24: 0x000000010c5ccbc3 WordPress`@objc QuickStartNavigationWatcher.navigationController(_:willShow:animated:) at <compiler-generated>:0
    frame #25: 0x000000010c908dc8 WordPress`partial apply at <compiler-generated>:0
    frame #26: 0x000000010c907361 WordPress`thunk for @escaping @callee_guaranteed (@unowned UINavigationController, @unowned UIViewController, @unowned Bool) -> () at <compiler-generated>:0
    frame #27: 0x000000010c908e26 WordPress`thunk for @escaping @callee_guaranteed (@unowned UINavigationController, @unowned UIViewController, @unowned Bool) -> ()partial apply at <compiler-generated>:0
    frame #28: 0x000000010c907309 WordPress`NavigationControllerDelegateRepeater.navigationController(navigationController=0x00007fd8898eb400, viewController=0x00007fd8884beb90, animated=false, self=0x000060000327cd10) at ManyDelegateNavigationController.swift:18
    frame #29: 0x000000010c9073d3 WordPress`@objc NavigationControllerDelegateRepeater.navigationController(_:willShow:animated:) at <compiler-generated>:0
    frame #30: 0x000000011b82f7a6 UIKitCore`-[UINavigationController _startTransition:fromViewController:toViewController:] + 1089
    frame #31: 0x000000011b83048b UIKitCore`-[UINavigationController _startDeferredTransitionIfNeeded:] + 896
    frame #32: 0x000000011b8317e0 UIKitCore`-[UINavigationController __viewWillLayoutSubviews] + 150
    frame #33: 0x000000011b811600 UIKitCore`-[UILayoutContainerView layoutSubviews] + 217
    frame #34: 0x000000013a8a86c4 UIKit`-[UILayoutContainerViewAccessibility layoutSubviews] + 42
    frame #35: 0x000000011c3d8795 UIKitCore`-[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1441
    frame #36: 0x0000000113a54b19 QuartzCore`-[CALayer layoutSublayers] + 175
    frame #37: 0x0000000113a599d3 QuartzCore`CA::Layer::layout_if_needed(CA::Transaction*) + 395
    frame #38: 0x00000001139d27ca QuartzCore`CA::Context::commit_transaction(CA::Transaction*) + 342
    frame #39: 0x0000000113a0997e QuartzCore`CA::Transaction::commit() + 576
    frame #40: 0x0000000113a0a6fa QuartzCore`CA::Transaction::observer_callback(__CFRunLoopObserver*, unsigned long, void*) + 76
    frame #41: 0x0000000117f4bc27 CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
    frame #42: 0x0000000117f460be CoreFoundation`__CFRunLoopDoObservers + 430
    frame #43: 0x0000000117f46751 CoreFoundation`__CFRunLoopRun + 1537
    frame #44: 0x0000000117f45e11 CoreFoundation`CFRunLoopRunSpecific + 625
    frame #45: 0x000000011ee471dd GraphicsServices`GSEventRunModal + 62
    frame #46: 0x000000011beee81d UIKitCore`UIApplicationMain + 140
    frame #47: 0x000000010bd94891 WordPress`main(argc=2, argv=0x00007ffee3e713e0) at main.m:11
    frame #48: 0x0000000119384575 libdyld.dylib`start + 1
```

Same or similar crash: 5c0849a9f8b88c2963b031da-fabric

Marking as high priority since the crash in Fabric is fairly frequent (in the top 10 crashes) and it's a blocker for enabling the related UI test.

**Comment by loremattei** [comment]
@SergioEstevao @designsimply  It seems to be still happening in the latest 11.9.1.

**Comment by SergioEstevao** [comment]
The PR with the fix for this only targeted 12.0.
https://github.com/wordpress-mobile/WordPress-iOS/pull/11212
